plugins/web/opentelemetry-plugin-react-load/src/types.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+
+/*
+ * method "render" from React.Component
+ */
+export type RenderFunction = () => React.ReactNode;
+
+/*
+ * method "componentDidMount" from React.Component
+ */
+export type ComponentDidMountFunction = (() => void) | undefined;
```

Comment:
I had to add `| undefined` in order to match the function definition provided by React, otherwise when I tried to wrap a function using Shimmer it didn't compile since the types of the patch I'm supplying and actual function prototype passed in don't match